File: 2022/largestContainer.py
def maxArea(self, heights):
        """
        :type heights: List[int]
        :rtype: int
        """
        max_area = 0
        l = 0
        r = len(heights) - 1
        
        while(r > l):
            dist = r - l
            height = min(heights[l], heights[r])
            area = dist * height
            max_area = area if area > max_area else max_area
 
            if heights[l] > heights[r]:
                r -= 1
            else:
                l += 1
                
        return max_area

File: 2022/largestContainer.js
// Find two lines that together with the x-axis form a container, such that the container contains the most water.

let maxArea = (arr) => {
    
    let l = 0;
    let r = arr.length - 1;
    let maxArea = 0;
    
    // create bounding box w/ most area
    while (l != r || l < r) {
        const left = arr[l];
        const right = arr[r];
        const dist = r - l;
        
        if (left < right) {
            maxArea = left * dist > maxArea ? left * dist : maxArea;
            l++;
        } else {
            maxArea = right * dist > maxArea ? right * dist : maxArea;
            r--;
        }
    }
    
    return maxArea;
};